core/services/pipeline/task.mode_test.go

```diff
@@ -112,87 +112,91 @@ func TestModeTask(t *testing.T) {
 		})
 	}
 
-	for _, test := range tests {
-		test := test
-		t.Run(test.name+" (VarExpr)", func(t *testing.T) {
-			t.Parallel()
+	t.Run("VarExpr", func(t *testing.T) {
+		for _, test := range tests {
+			test := test
+			t.Run(test.name, func(t *testing.T) {
+				t.Parallel()
 
-			var inputs []interface{}
-			for _, input := range test.inputs {
-				if input.Error != nil {
-					inputs = append(inputs, input.Error)
+				var inputs []interface{}
+				for _, input := range test.inputs {
+					if input.Error != nil {
+						inputs = append(inputs, input.Error)
+					} else {
+						inputs = append(inputs, input.Value)
+					}
+				}
+				vars := pipeline.NewVarsFrom(map[string]interface{}{
+					"foo": map[string]interface{}{"bar": inputs},
+				})
+				task := pipeline.ModeTask{
+					BaseTask:      pipeline.NewBaseTask(0, "task", nil, nil, 0),
+					Values:        "$(foo.bar)",
+					AllowedFaults: test.allowedFaults,
+				}
+				output := task.Run(context.Background(), vars, nil)
+				if output.Error != nil {
+					require.Equal(t, test.wantErrorCause, errors.Cause(output.Error))
+					require.Nil(t, output.Value)
 				} else {
-					inputs = append(inputs, input.Value)
+					require.Equal(t, map[string]interface{}{
+						"results":     test.wantResults,
+						"occurrences": test.wantOccurrences,
+					}, output.Value)
+					require.NoError(t, output.Error)
 				}
-			}
-			vars := pipeline.NewVarsFrom(map[string]interface{}{
-				"foo": map[string]interface{}{"bar": inputs},
 			})
-			task := pipeline.ModeTask{
-				BaseTask:      pipeline.NewBaseTask(0, "task", nil, nil, 0),
-				Values:        "$(foo.bar)",
-				AllowedFaults: test.allowedFaults,
-			}
-			output := task.Run(context.Background(), vars, nil)
-			if output.Error != nil {
-				require.Equal(t, test.wantErrorCause, errors.Cause(output.Error))
-				require.Nil(t, output.Value)
-			} else {
-				require.Equal(t, map[string]interface{}{
-					"results":     test.wantResults,
-					"occurrences": test.wantOccurrences,
-				}, output.Value)
-				require.NoError(t, output.Error)
-			}
-		})
-	}
+		}
+	})
```

Comment:
These are actually clean indents. Wrapping each loop up in a non-parallel subtest keeps the inner parallel tests isolated to their group. This prevents a `VarExpr` and `JSONWithVarExprs` subtest for the same test definition from racing for access.
This is 'fixed' in that it is no longer racing, but I can't explain why there were _writes_ to the test definition in the first place.